### PR TITLE
[CI][Tests][gfx90a] Basic support for gfx90a (partially implements #1031)

### PR DESCRIPTION
**Pull request by junliume**
[Target]
Test support for gfx90a to (completely) mimic gfx908  (#1031)

[WIP]
PR #1066 is in progress and this PR has not included the changes made in it

### CODE REVIEW

**Comment by atamazov** [comment]
FYI in order to mimic gfx908 it is enough to pass -DMIOPEN_TEST_GFX908=On to cmake.

**Comment by atamazov** [comment]
I think this should implement (3) from https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1031#issuecomment-889725745. Therefore Jenkinsfile changes should be reverted.

**Comment by shaojiewang** [comment]
> I think this should implement (3) from [#1031 (comment)](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1031#issuecomment-889725745). Therefore Jenkinsfile changes should be reverted.

@atamazov Jenkinsfile is moved to #1082

**Comment by junliume** [comment]
Trying to merge this one first, so I have rebased and resolved conflicts.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1152,7 +1161,7 @@ COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --
 COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input 128  256 56 56 --weights 64  256 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-forward --disable-backward-weights
 )
 
-add_custom_test(test_conv_igemm_dynamic_xdlops_bwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED VEGA_DISABLED
+add_custom_test(test_conv_igemm_dynamic_xdlops_bwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED GFX90a_ENABLED VEGA_DISABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1164]: this test does not support `gfx90a`

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1167,7 +1176,7 @@ COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIO
 COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  8  16 5 5 --weights 8  16  2 2 --pads_strides_dilations 0 0 1 1 1 1 --disable-forward --disable-backward-weights
 )
 
-add_custom_test(test_conv_igemm_dynamic_xdlops_bwd_group SKIP_UNLESS_ALL HALF_ENABLED FLOAT_DISABLED GFX908_ENABLED VEGA_DISABLED
+add_custom_test(test_conv_igemm_dynamic_xdlops_bwd_group SKIP_UNLESS_ALL HALF_ENABLED FLOAT_DISABLED GFX908_ENABLED GFX90a_ENABLED VEGA_DISABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1179]: this test does not support `gfx90a`

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -1177,7 +1186,7 @@ add_custom_test(test_conv_igemm_dynamic_xdlops_bwd_float SKIP_UNLESS_ALL HALF_DI
 COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  4  512 128 128 --weights 12  512  1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-forward --disable-backward-weights
 )
 
-add_custom_test(test_conv_igemm_dynamic_xdlops_fwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED VEGA_DISABLED
+add_custom_test(test_conv_igemm_dynamic_xdlops_fwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED GFX90a_ENABLED VEGA_DISABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1189]: this test does not support `gfx90a`

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -1197,7 +1206,7 @@ COMMAND ${DYNAMIC_IMPLICITGEMM_FWD_GTC_DYNAMIC_XDLOPS_ENVS} $<TARGET_FILE:test_c
 COMMAND ${DYNAMIC_IMPLICITGEMM_FWD_GTC_DYNAMIC_XDLOPS_ENVS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input 64 3 230 230 --weights 64 3 7 7 --pads_strides_dilations 0 0 2 2 1 1 --disable-backward-data --disable-backward-weights
 )
 
-add_custom_test(test_conv_igemm_dynamic_xdlops_wrw SKIP_UNLESS_ALL GFX908_ENABLED VEGA_DISABLED HALF_ENABLED
+add_custom_test(test_conv_igemm_dynamic_xdlops_wrw SKIP_UNLESS_ALL GFX908_ENABLED GFX90a_ENABLED VEGA_DISABLED HALF_ENABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1209]: this test does not support `gfx90a`

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -1224,7 +1233,7 @@ COMMAND ${DYNAMIC_IMPLICITGEMM_WRW_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIO
 COMMAND ${DYNAMIC_IMPLICITGEMM_WRW_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  1 128 56 56 --weights 1 128 5 5 --pads_strides_dilations 0 0 2 2 1 1 --disable-forward --disable-backward-data
 )
 
-add_custom_test(test_conv_igemm_dynamic_xdlops_nhwc_fwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED VEGA_DISABLED
+add_custom_test(test_conv_igemm_dynamic_xdlops_nhwc_fwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED GFX90a_ENABLED VEGA_DISABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1236]: this test only support `gfx90a` till https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1035 is merged.
cc @shaojiewang 
If this PR should be merged first, I think this test enable 90a better be added in #1035

File: test/CMakeLists.txt [hunk 6]
```diff
@@ -1241,7 +1250,7 @@ COMMAND ${DYNAMIC_IMPLICITGEMM_XDLOPS_NHWC_FWD_ENVS} $<TARGET_FILE:test_conv2d>
 COMMAND ${DYNAMIC_IMPLICITGEMM_XDLOPS_NHWC_FWD_ENVS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  64   3 78 78 --weights  64   3 7 7 --pads_strides_dilations 0 0 2 2 1 1 --disable-backward-data --disable-backward-weights --in_layout NHWC --fil_layout NHWC --out_layout NHWC
 COMMAND ${DYNAMIC_IMPLICITGEMM_XDLOPS_NHWC_FWD_ENVS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  16 192 17 17 --weights 224 192 1 7 --pads_strides_dilations 0 3 1 1 1 1 --disable-backward-data --disable-backward-weights --in_layout NHWC --fil_layout NHWC --out_layout NHWC
 )
-add_custom_test(test_conv_igemm_dynamic_xdlops_nhwc_bwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED VEGA_DISABLED
+add_custom_test(test_conv_igemm_dynamic_xdlops_nhwc_bwd SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED GFX90a_ENABLED VEGA_DISABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1253]: this test only support `gfx90a` till #1035 is merged.

File: test/CMakeLists.txt [hunk 7]
```diff
@@ -1269,7 +1278,7 @@ set(ARGS_NHWC_WRW
     --fil_layout NHWC
     --out_layout NHWC)
 
-add_custom_test(test_conv_igemm_dynamic_xdlops_nhwc_wrw SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED VEGA_DISABLED
+add_custom_test(test_conv_igemm_dynamic_xdlops_nhwc_wrw SKIP_UNLESS_ALL HALF_ENABLED GFX908_ENABLED GFX90a_ENABLED VEGA_DISABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1281]: this test only support `gfx90a` till #1035 is merged.

File: test/CMakeLists.txt (hunk 2 above)
Comment (junliume) [reply test/CMakeLists.txt]: removed as suggested

File: test/CMakeLists.txt (hunk 3 above)
Comment (junliume) [reply test/CMakeLists.txt]: ditto

**Comment by junliume** [comment]
Mission consolidated into #1082

**Pull request closed without merging**